mmv1/api/resource.rb

```diff
@@ -435,6 +439,34 @@ def decoder?
       !@transport&.decoder.nil?
     end
 
+    def add_labels_related_fields(props)
+      props.each do |p|
+        if p.is_a? Api::Type::KeyValueLabels
+          props << build_effective_field('labels', p.field_min_version)
+        elsif p.is_a? Api::Type::KeyValueAnnotations
+          props << build_effective_field('annotations', p.field_min_version)
+        elsif (p.is_a? Api::Type::NestedObject) && !p.all_properties.nil?
+          p.properties = add_labels_related_fields(p.all_properties)
+        end
+      end
+      props
+    end
+
+    def build_effective_field(name, min_version)
```

Comment:
nit: Consider scoping this a little more, maybe `build_effective_labels_field` or `build_effective_kvp_field`?

Reply:
I will change it to `build_effective_labels_field`